### PR TITLE
[flang][fir][NFC] Move ComplexType to TableGen type definition

### PR DESCRIPTION
**Pull request by clementval**
New PR after rebase of `fir-dev`. Original PR #614

### CODE REVIEW

File: flang/include/flang/Optimizer/Dialect/FIRTypes.td [hunk 1]
```diff
@@ -21,6 +21,32 @@ class FIR_Type<string name, string typeMnemonic> : TypeDef<fir_Dialect, name> {
...
+def fir_ComplexType : FIR_Type<"Complex", "complex"> {
+  let summary = "Complex type";
+
+  let description = [{
+    Model of a Fortran COMPLEX intrinsic type, including the KIND type
+    parameter. COMPLEX is a floating point type with a real and imaginary
+    member.
+  }];
+
+  let parameters = (ins "KindTy":$kind);
```
Comment (schweitzpgi) [inline flang/include/flang/Optimizer/Dialect/FIRTypes.td:33]: Were you going to rename this `fKind`?

**Review by schweitzpgi — APPROVED** [review]
+1

File: flang/include/flang/Optimizer/Dialect/FIRTypes.td (hunk 1 above)
Comment (clementval) [reply flang/include/flang/Optimizer/Dialect/FIRTypes.td:33]: I can update it if it makes more sense so there will be only one accessor in the end. What do you think?

**Review by jeanPerier — APPROVED** [review]

File: flang/include/flang/Optimizer/Dialect/FIRTypes.td (hunk 1 above)
Comment (schweitzpgi) [reply flang/include/flang/Optimizer/Dialect/FIRTypes.td:33]: Makes sense to just have one.

Comment (clementval) [reply flang/include/flang/Optimizer/Dialect/FIRTypes.td:33]: Ok I'll update it.

**Pull request merged**